app/src/main/java/com/alexbar/layoutbasic/utils/MoviesData.kt

```diff
@@ -0,0 +1,63 @@
+package com.alexbar.layoutbasic.utils
+
+import android.content.Context
+import com.google.gson.Gson
+import com.google.gson.annotations.SerializedName
+import com.google.gson.reflect.TypeToken
+import java.io.IOException
+
+fun getJsonDataFromAsset(
+    context: Context,
+    fileName: String
+): String? {
+    val jsonString: String
+    try {
+        jsonString = context.assets.open(fileName).bufferedReader().use {
+            it.readText()
+        }
+    } catch (exp: IOException) {
+        exp.printStackTrace()
+        return null
+    }
+
+    return jsonString
+}
+
+fun movieList(context: Context): MutableList<Movie> {
+    val jsonFileString = getJsonDataFromAsset(context = context, "movies.json")
+    val type = object : TypeToken<List<Movie>>() {}.type
+    return Gson().fromJson(jsonFileString, type)
+}
+
+fun tvSeriesList(context: Context): MutableList<TVSeries> {
+    val jsonFileString = getJsonDataFromAsset(context = context, "series.json")
+    val type = object : TypeToken<List<TVSeries>>() {}.type
+    return Gson().fromJson(jsonFileString, type)
+}
+
+sealed class Media
+data class Movie(
+    @SerializedName("title")
+    val title: String,
+    @SerializedName("overview")
+    val description: String,
+    @SerializedName("poster_path")
+    val imageUrl: String,
+    @SerializedName("first_air_date")
+    val firstAirDate: String,
+    @SerializedName("vote_average")
+    val voteAverage: Double
+) : Media()
+
+data class TVSeries(
+    @SerializedName("name")
+    val name: String,
+    @SerializedName("overview")
+    val description: String,
+    @SerializedName("poster_path")
+    val imageUrl: String,
+    @SerializedName("first_air_date")
+    val firstAirDate: String,
+    @SerializedName("vote_average")
+    val voteAverage: Double
+): Media()
```

Comment:
@ezatpanah I declared these classes as sealed classes, but I'm wondering if there might be a better way to implement them, since both classes share the same fields, with the only difference being that in one class the field is called 'name' and in the other 'title'

Reply:
Using a sealed class here is quite fitting as it allows you to represent a restricted hierarchy of classes, ensuring that you handle all possible cases.

I can suggest a better way to handle this. I don't see any difference between your two data classes(just the name and title is different). So, maybe we can simplify the code to make it shorter and easier to read.

first, you can have an enum class :

```
enum class MediaType {
    MOVIE,
    TV_SERIES
}
```
and then use it in your single data class like :

```
data class Media(
    val title: String,
    val description: String,
    val imageUrl: String,
    val firstAirDate: String,
    val voteAverage: Double,
    val mediaType: MediaType
)
```